services/phish_service.py

```diff
@@ -74,6 +74,18 @@ def adjust_for_linkedin_url(user_input: str) -> str:
         return user_input[:-1]
     return user_input
 
+def get_all_phishing_email_requested_by_user(user_info: dict) -> list[dict]:
+    phishing_emails = DB.get_instance().get_previous_phishing_email_generated_by_user(user_info['email'])
+
+    for item in phishing_emails:
+        public_identifier = item.get('linkedin_data.public_identifier','')
+
+        if public_identifier not in PROFILE_IMAGES_CACHE:
+            profile_image = item.get('profile_image','')
+            PROFILE_IMAGES_CACHE[public_identifier] = profile_image
+
+    return phishing_emails
```

Comment:
I would remove the if check because for example if expiration of element is 295 seconds, you don't want that after 5 seconds the image is not available anymore, and remember one line more spacing before the function

Reply:
I had added the check because we are getting the data from the DB in reverse chronological order and the initial entries to the database did not have the user's profile picture. This means that for those user's the profile picture will always be blank.